inc/endpoints/class-post-select-controller.php

```diff
@@ -140,7 +140,12 @@ public function get_items( $request ) {
 
 		$query_args = apply_filters( 'hm_gb_tools_post_select_query_args', $query_args );
 
+		do_action( 'hm_gb_tools_post_select_before_query', $query_args );
+
 		$query = new WP_Query( $query_args );
+
+		do_action( 'hm_gb_tools_post_select_after_query', $query_args );
```

Comment:
I don't quite get why we'd need an `after` filter. What would you need to do based on the query arguments that couldn't be done by filtering them before the query is executed?

Reply:
I suppose @Rayhatron is wanting to `add_filter('posts_search', '__search_by_title_only', 500, 2);` and then clean up afterwards. Thats probably best practice, although given this is an API request, maybe not necessary. 

But could also use a filter for that too something like `the_posts` which is called after the posts are fetched.

Reply:
Yes, that's exactly what I plan to use the actions for. I'll look more into an alt implementation that uses the hm_gb_tools_post_select_query_args filter and doesn't require actions to be added to this repo. Just that when I was looking it earlier, I couldn't find a way to limit the returned posts to those that matched the search string in the title only.

Seemed modifying the query was the way to go hence the actions were added to add a posts_search filter before the query that modifies it and remove said filter after the query.

Reply:
Hey @mattheu @kadamwhite I've updated the implementation to use the existing hm_gb_tools_post_select_query_args filter.

A use case would then be to do this in a theme or plugin code outside of this plugin.

```php
/**
 * Modify hm_gb_tools_post_select_query_args to search by title only
 * Will limit the returned posts to those that matched the search string in title only.
 *
 * @return array
 */
function add_search_by_title_only( $query_args ): array {

	$query_args['search_by_title_only'] = true;

	return $query_args;
}
add_filter( 'hm_gb_tools_post_select_query_args', __NAMESPACE__ . '\\add_search_by_title_only', 1, 10 );
```